src/chrome/content/scaffold/templates/newWeb.js

```diff
@@ -23,29 +23,36 @@
 
 
 // attr()/text() v2
-function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null;}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null;}
-
+function attr(docOrElem, selector, attr, index) {
+	var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector);
+	return elem ? elem.getAttribute(attr) : null;
+}
+function text(docOrElem, selector, index) {
+	var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector);
+	return elem ? elem.textContent : null;
+}
 
 function detectWeb(doc, url) {
 	// TODO: adjust the logic here
 	if (url.includes('/article/')) {
 		return "newspaperArticle";
-	} else if (getSearchResults(doc, true)) {
+	}
+	else if (getSearchResults(doc, true)) {
 		return "multiple";
 	}
+	return false;
 }
 
-
 function getSearchResults(doc, checkOnly) {
 	var items = {};
 	var found = false;
 	// TODO: adjust the CSS selector
 	var rows = doc.querySelectorAll('h2>a.title[href*="/article/"]');
-	for (let i=0; i<rows.length; i++) {
+	for (var row of Array.from(rows)) {
```

Comment:
Is `Array.from` needed here?

Reply:
It is unfortunately; the result from `querySelectorAll` is a NodeList, which looks a little like an array (you can use brackets, and it has a `length`), but it's not actually an array, and for-of fails on it otherwise.

Reply:
Turns out it isn't actually (thanks @dstillman), NodeList is indeed iterable.